[mssqlserver] Add all SP cycles

BiNZGi opened the pull request:
https://learn.microsoft.com/lifecycle/products/?terms=SQL%20Server
https://learn.microsoft.com/troubleshoot/sql/releases/download-and-install-latest-updates
Relates to #3960

BiNZGi committed:
[mssqlserver] Update releaseDate to RTM date

https://learn.microsoft.com/lifecycle/products/?terms=SQL%20Server
Relates to #3960

BiNZGi committed:
[mssqlserver] Add SP cycles for 2005, 2008, 2008R2

BiNZGi committed:
.

BiNZGi committed:
Merge branch 'master' into mssqlserver

BiNZGi committed:
.

BiNZGi committed:
.

BiNZGi committed:
.

BiNZGi committed:
.

BiNZGi commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,415 @@ versionCommand: select @@version
     extendedSupport: false
-    latest: "16.0.4085.2"
+    latestGdr: "16.0.1105.1"
+    latestGdrLink: https://support.microsoft.com/help/5029379
```
Preview: https://deploy-preview-3992--endoflife-date.netlify.app/mssqlserver
I wanted to have a link on the custom column entry.
I didn't find a way in https://endoflife.date/contribute
The table is too wide for the web view and the user has to scroll to get to the Latest CU column.
@marcwrobel any thoughts?

BiNZGi replied:
To keep this MR simpler I have removed the custom column and I'll file a separate MR for that.

BiNZGi committed:
[mssqlserver] separate custom column into own MR

BiNZGi committed:
[mssqlserver] Update 2022 CU10

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
+# releaseDate/support/eol/extendedSupport: https://learn.microsoft.com/lifecycle/products/?terms=SQL%20Server
+# Overview over all versions: https://learn.microsoft.com/troubleshoot/sql/releases/download-and-install-latest-updates
 releases:
 -   releaseCycle: "2022"
```
Should we use 16 as the release cycles, and 2022 as the release label ?

BiNZGi replied:
Update done see f093bdaeb242264dd7445e3b3ec555d31253deac.

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
-    latest: "14.0.3465.1"
-    latestReleaseDate: 2023-10-10
-    link: https://support.microsoft.com/help/5029376
+    latestGdr: "16.0.1105.1"
```
Are we sure we want to document also the GDR versions ? Do you know if it's common to use those GDR versions ?

BiNZGi replied:
Yes, it is common. Companies only apply GDR updates for certain servers to have minimal impact but have security fixes. CU and GDR are explained on L444 resp. L446

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
-    link: https://support.microsoft.com/help/5029376
+    latestGdr: "16.0.1105.1"
+    latestGdrLink: https://support.microsoft.com/help/5029379
+    latest: "16.0.4095.4 CU10"
```
Looking at https://learn.microsoft.com/en-us/troubleshoot/sql/releases/download-and-install-latest-updates#sql-server-2022 I think it could be better to track the CU in an individual field. It feels more natural to only have the version in the latest field, at least for API users. WDYT ?

We may miss a _latestReleaseLabel_ feature to display all those informations though ?

BiNZGi replied:
That's a good point I see three options:
1. Block this PR until we have the feature `latestReleaseLabel`
2. Remove the short terms in latest until we have the feature `latestReleaseLabel`
3. Leave as it is now and update when we have the feature `latestReleaseLabel`

I vote for option 3.

marcwrobel replied:
Ok, let's start with option 3.

BiNZGi replied:
Created #4196 for the new feature `latestReleaseLabel`

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
 extendedSupportColumn: Extended Security Updates
 
+# releaseDate/support/eol/extendedSupport: https://learn.microsoft.com/lifecycle/products/?terms=SQL%20Server
+# Overview over all versions: https://learn.microsoft.com/troubleshoot/sql/releases/download-and-install-latest-updates
```
I think we can set up an automation with this.

BiNZGi replied:
I would handle that outside of this PR.

marcwrobel reviewed (commented):
Did not check the dates yet, but here are my first thoughts.

BiNZGi committed:
[mssqlserver] Update releaseCycle to base version

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
+    codename: SQL16
+    releaseLabel: "2016 SP2"
+    releaseDate: 2018-04-24
+    support: 2022-10-11
```
Why not 2021-07-13 as for 2016 SP3 ?

BiNZGi replied:
SP2 has an earlier end date than SP3, see https://learn.microsoft.com/lifecycle/products/sql-server-2016
> Releases, Entry Service Pack 2 -> End Date: Oct 11, 2022

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
     codename: Denali
     releaseLabel: "2012 SP4"
     releaseDate: 2017-10-05
     support: 2017-07-11
```
End of support is before release date !?

BiNZGi replied:
Fixed with 9ac554fd79012bf22e185bbfa4fc2be165097e3b

marcwrobel reviewed (commented):
Reviewed up to 2017 (Helsinki) so far. Here are my first comments.

BiNZGi committed:
[mssqlserver] Update 11.0-sp4 support

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
+    codename: Hekaton
+    releaseLabel: "2014 SP2"
+    releaseDate: 2016-07-14
+    support: 2020-01-14
```
Should be Jul 9, 2019, as per https://learn.microsoft.com/en-us/lifecycle/products/sql-server-2014 ?

BiNZGi replied:
Careful, on https://learn.microsoft.com/en-us/lifecycle/products/sql-server-2014 see Section "Releases", Entry "Service Pack 2" -> Column "End Date" -> Jan 14, 2020

marcwrobel replied:
I thought the support date was using either the Mainstream End Date, or the Release End Date if it was earlier. If I understand correctly this is only true for the latest service pack (a few SP3 are in this case) ?

marcwrobel replied:
And in this case is it correct to consider that the _Extended Security Updates_ will be applied to the last service pack ?

BiNZGi replied:
It is the "End Date" for the releaseCycle (with the SP applied).

> And in this case is it correct to consider that the Extended Security Updates will be applied to the last service pack ?

Yes, that is correct.

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
+    latestGdr: "12.0.5223.6"
+    latestGdrLink: https://support.microsoft.com/help/4505217
+    latest: "12.0.5687.1 CU18"
+    latestReleaseDate: 2019-07-29
```
I only found the month and year. Where can I confirm the day ?

BiNZGi replied:
On https://support.microsoft.com/help/4500180, see Link "Download the latest cumulative update package for SQL Server 2014 SP2 now."

Direct link is: https://www.microsoft.com/download/details.aspx?id=53592

> Date Published: 7/29/2019

marcwrobel reviewed (commented):
Reviewed until -   2016 (SQL16), here are a few more comments.

BiNZGi committed:
[mssqlserver] Update 2019 CU24

https://learn.microsoft.com/troubleshoot/sql/releases/sqlserver-2019/cumulativeupdate24

marcwrobel commented on products/mssqlserver.md:
```diff
@@ -11,135 +11,408 @@ versionCommand: select @@version
+-   releaseCycle: "10.50-sp3"
     codename: Kilimanjaro
     releaseLabel: "2008 R2 SP3"
     releaseDate: 2014-09-26
```
```suggestion
    releaseDate: 2014-07-07
```

According to https://learn.microsoft.com/en-us/lifecycle/products/microsoft-sql-server-2008-r2.

BiNZGi committed:
Update 2022 CU11

BiNZGi committed:
Update 2008 R2 SP3 releaseDate

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

BiNZGi committed:
[mssqlserver] Update 2019 CU25

https://learn.microsoft.com/troubleshoot/sql/releases/sqlserver-2019/cumulativeupdate25

BiNZGi committed:
[mssqlserver] Update 2022 CU12 

https://learn.microsoft.com/troubleshoot/sql/releases/sqlserver-2022/cumulativeupdate12

BiNZGi committed:
[mssqlserver] Update 2022 GDR, 2022 CU12+GDR, 2019 GDR, 2019 CU25+GDR

https://learn.microsoft.com/en-us/troubleshoot/sql/releases/download-and-install-latest-updates#latest-updates-available-for-currently-supported-versions-of-sql-server
2022 GDR https://support.microsoft.com/help/5035432
2022 CU12+GDR https://support.microsoft.com/help/5036343
2019 GDR https://support.microsoft.com/help/5035434
2019 CU25+GDR https://support.microsoft.com/help/5036335

BiNZGi committed:
[mssqlserver] 2019 CU26

https://learn.microsoft.com/troubleshoot/sql/releases/sqlserver-2019/cumulativeupdate26

BiNZGi committed:
[mssqlserver] Update 2022 CU13

https://learn.microsoft.com/troubleshoot/sql/releases/sqlserver-2022/cumulativeupdate13

BiNZGi committed:
Merge branch 'master' into mssqlserver

BiNZGi committed:
.

BiNZGi committed:
.

BiNZGi committed:
.

BiNZGi committed:
.

BiNZGi committed:
[mssqlserver] 2022 CU13 GDR, 2022 GDR, 2019 CU27 GDR, 2019 GDR, 2017 CU31 GDR, 2017 GDR, 2016 SP3 GDR

2022 CU13 GDR -> https://support.microsoft.com/help/5040939
2022 GDR -> https://support.microsoft.com/help/5040936
2019 CU27 GDR -> https://support.microsoft.com/help/5040948
2019 GDR -> https://support.microsoft.com/help/5040986
2017 CU31 GDR -> https://support.microsoft.com/help/5040940
2017 GDR -> https://support.microsoft.com/help/5040942
2016 SP3 GDR -> https://support.microsoft.com/help/5040946

BiNZGi committed:
[mssqlserver] Update 2014 eoes

https://learn.microsoft.com/lifecycle/faq/extended-security-updates

BiNZGi committed:
[mssqlserver] Update 2022 CU14

https://learn.microsoft.com/troubleshoot/sql/releases/sqlserver-2022/cumulativeupdate14

BiNZGi committed:
[mssqlserver] Update 2019 CU28

https://learn.microsoft.com/troubleshoot/sql/releases/sqlserver-2019/cumulativeupdate28

BiNZGi committed:
[mssqlserver] Update 2022 CU15

https://learn.microsoft.com/troubleshoot/sql/releases/sqlserver-2022/cumulativeupdate15

BiNZGi committed:
[mssqlserver] Update latest

https://learn.microsoft.com/sql/releases/download-and-install-latest-updates

BiNZGi committed:
Merge branch 'master' into mssqlserver

BiNZGi committed:
[mssqlserver] Update latest

BiNZGi committed:
Merge branch 'master' into mssqlserver

BiNZGi committed:
Merge branch 'master' into mssqlserver

BiNZGi committed:
[mssqlserver] Fixes
